--- rootfs/usr/local/bin/lib/pdu_tools.rb
+++ rootfs/usr/local/bin/lib/pdu_tools.rb
@@ -1,4 +1,3 @@
-require 'phone'
 require 'active_support/all'
 
 require_relative './pdu_tools/helpers'
--- rootfs/usr/local/bin/lib/pdu_tools/encoder.rb
+++ rootfs/usr/local/bin/lib/pdu_tools/encoder.rb
@@ -100,13 +100,10 @@
     end
 
     def prepare_recipient recipient
-      Phoner::Phone.default_country_code ||= "421"
+
+
       address_type = "91" # International
-      if @options[:force_recipient]
-        address = recipient
-      else
-        address = Phoner::Phone.parse(recipient).format("%c%a%n")
-      end
+      address = recipient
       address_length = "%02X" % address.length
       address_encoded = normal2swapped address
       address_length + address_type + address_encoded
--- rootfs/usr/local/bin/droid4-sms-send.rb
+++ rootfs/usr/local/bin/droid4-sms-send.rb
@@ -7,7 +7,7 @@
 #
 
 require "mail"
-require "pdu_tools"
+require_relative "lib/pdu_tools"
 
 $device = "motmdm3"
 
--- rootfs/usr/local/bin/droid4-sms-read.rb
+++ rootfs/usr/local/bin/droid4-sms-read.rb
@@ -12,12 +12,12 @@
 # with maildirmake ~/Maildir/INBOX.sms
 #

-require "pdu_tools"
+require_relative "lib/pdu_tools"

 $device = "motmdm9"
-$maildir = "#{Dir.home}/Maildir/INBOX.sms"
+$maildir = "/root/Maildir/INBOX.sms"
 $arg0 = ARGV[0]

 def print_help()
   printf "Usage: %s [optionalpdu]\n", $0
   printf "Without options, %s keeps reading /dev/motmdm9, then\n", $0
@@ -68,7 +68,7 @@
     printf "File already exists for %s\n", filename
     return -1
   end
-
+  Process.spawn("pwm-vibrator.py")
   return decoded.size
 end

