--- overlays/lib/pdu_tools.rb	2018-12-03 15:44:38.000000000 +0200
+++ overlays/lib/pdu_tools.rb	2019-01-17 22:16:58.662588563 +0200
@@ -1,4 +1,3 @@
-require 'phone'
 require 'active_support/all'
 
 require_relative './pdu_tools/helpers'
--- overlays/lib/pdu_tools/encoder.rb	2018-12-03 15:44:38.000000000 +0200
+++ overlays/lib/pdu_tools/encoder.rb	2019-01-17 22:26:44.850986354 +0200
@@ -100,13 +100,10 @@
     end
 
     def prepare_recipient recipient
-      Phoner::Phone.default_country_code ||= "421"
+
+
       address_type = "91" # International
-      if @options[:force_recipient]
-        address = recipient
-      else
-        address = Phoner::Phone.parse(recipient).format("%c%a%n")
-      end
+      address = recipient
       address_length = "%02X" % address.length
       address_encoded = normal2swapped address
       address_length + address_type + address_encoded
